README.md

```diff
@@ -1,117 +1,209 @@
-####  Sign in
-Once you signed up, you can either click on the Todo tab or the sign in link to gain access to the application. Note that if you are doing this in the same session in which you signed up, you will automatically sign in with the same account you used for signing up. If you are signing in during a new session, you will be presented with Azure AD's credentials prompt: sign in using an account compatible with the sign up option you chose earlier (the exact same account if you used user consent, any user form the same tenant if you used admin consent). 
+If the app is not provisioned in your tenant by the tenant administrator using the steps laid out above, the sign-up process will result in the following error.
 
-## How To Deploy This Sample to Azure
+![Need admin approval](./ReadmeFiles/NeedAdminApproval.JPG)
 
-Coming soon.
+This step uses the `prompt=admin_consent' option provided in the [OAuth 2.0 authorization code grant](https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-protocols-oauth-code#request-an-authorization-code) to provide the administrator an option to consent for the entire tenant.
 
-## Azure Government Deviations
+You can also use the [admin consent endpoint](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#using-the-admin-consent-endpoint) to provision the app in the chosen tenant.
 
-In order to run this sample on Azure Government you can follow through the steps above with a few variations:
+#### Sign in
+
+Once you signed up, you can either click on the `Todo` tab or the `sign in` link to gain access to the application. Note that if you are selecting **sign in** in the same session in which you signed up, you will automatically sign in with the same account you used for signing up.
+
+If you are signing in during a new session, you will be presented with Azure AD's credentials prompt: sign in using an account compatible with the sign up option you chose earlier (the exact same account if you used user consent, any user form the same tenant if you used admin consent).
+
+If you try to sign-in before the tenant administrator has provisioned the app in the tenant using the `Sign up` link above, you will see the following error.
+
+![AADSTS700016, App not found](./ReadmeFiles/AADSTS700016.JPG)
+
+## National Cloud Deviations
+
+In order to run this sample on a National Cloud, you can follow through the steps above with a few variations:
+
+1. You must register this sample for your Azure AD Tenant in a [National Clouds](https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud) by following [[Step 2]](#step-2-register-the-sample-with-your-azure-active-directory-tenant) above in the National cloud of your choice.
```

Comment:
the step 2 for registering the app here can not be used in case of National Cloud section because we do not have App Reg (preview) blade in sovereign cloud portals neither do we support the MSA scenarios. 
So the whole section needs to be changed according to (Azure gov).

Reply:
fixed